### PR TITLE
Add Option to Use Host Network and Configure Ports

### LINKED ISSUE #2750: Failed to create fleet
I installed agones with helm chart. I got the error when creating fleet as below:

**What happened**:
```
Error from server (InternalError): error when creating "fleet.yaml": Internal error occurred: failed calling webhook "mutations.agones.dev": failed to call webhook: Post "https://agones-controller-service.agones-system.svc:443/mutate?timeout=10s": Address is not allowed
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Agones version: 1.26.0
- Kubernetes version (use `kubectl version`): 

> Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-03T13:36:49Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.13-eks-15b7512", GitCommit:"94138dfbea757d7aaf3b205419578ef186dd5efb", GitTreeState:"clean", BuildDate:"2022-08-31T19:15:48Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: EKS
- Install method (yaml/helm): helm
- Troubleshooting guide log(s):
- Others:

### PR DESCRIPTION
**Pull request by Orza**
**What type of PR is this?**
/kind feature

**What this PR does / Why we need it**:

This PR enables the use of the host network and allows changing the port number of the Agones controller and Agones extensions.
This change is necessary for certain managed Kubernetes clusters (e.g., AWS EKS) that use custom CNI plugins (such as Calico or Cilium). These environments require this adjustment because the AWS-managed control plane cannot communicate (e.g. webhooks) with pod IP CIDRs.

Which issue(s) this PR fixes:

Closes #2750 

**Special notes for your reviewer**:

Tested on AWS EKS with Cilium CNI plugin.

### CODE REVIEW

**Comment by markmandel** [comment]
We had the same conversation on the attached ticket. I'd like to understand why EKS can't support the standard K8s webhook setup. Requiring hostNetwork for Admission Controllers and/or APIService seems like a misconfiguration / issue with EKS that should be fixed at the EKS level.

Is there any documentation stating that you _have_ to use hostNetwork for Admission Controllers?

**Comment by unlightable** [comment]
> I'd like to understand why EKS can't support the standard K8s webhook setup.
It's indeed an issue of cluster networking setup. But rather popular one if you want to switch from ENI (and the likes) on EKS: https://github.com/cilium/cilium/issues/21959

It stems from inability of control plane to reach in-cluster pod endpoints when data plane runs in separate network layer. And inability to actually reconfigure `kube-apiserver` as it is an AWS managed component.
And all available solutions are rather ugly:
- Staying on ENI is not an option as moving away from is is the purpose of rolling out different CNI.
- Exposing though LoadBalancer/NodePort service (in place of ClusterIP) seems to not help as kubernetes API aggregation layer extracts endpoints from it and goes there directly and it still is unable to reach inside separate network layer.
- hostNetwok is the least bad here

**Comment by markmandel** [comment]
Thanks for providing the links - this is starting to make more sense.

Once question though, I'm looking at the Cillium docs (after going through the PRs to find it):

https://docs.cilium.io/en/latest/installation/k8s-install-helm/#:~:text=Deploy%20Cilium%20release%20via%20Helm%3A

It does say (emphasis mine)

> The EKS API Server is unable to route packets to the overlay network. This implies that any webhook which needs to be accessed must be host networked **or exposed through a service** or ingress.

Are the docs wrong? Shouldn't a Service still work?

**Comment by unlightable** [comment]
> Shouldn't a Service still work?

It works if you run `kube-apiserver` with `--enable-aggregator-routing=true`. Then it won't "optimize" service exposing controller pods down to endpoints.
It also works in a sense if you front your service through another one in a reverse-proxy setup, and that proxy could be `hostNetwork` one.
There probably is a way to abuse ExternalIP services too, but it all seems much more tedious than rolling a `hostNetwork` agones controller. Moreso since `extensions` controller is split from the main one?

**Comment by unlightable** [comment]
They also refer to webhooks (e.g. `MutatingWebhookConfiguration`) that have an option to provide url as a target. And that url could be an external ip of loadbalanced service.

You don't have such luxury with `APIService` that has to reference a service inside the cluster.

**Comment by Orza** [comment]
Also I can reffer to this issues:
aws/containers-roadmap#2227
https://github.com/cilium/cilium/issues/30111

File: cmd/controller/main.go [hunk 1]
```diff
@@ -307,6 +311,7 @@ func parseEnvFlags() config {
 	pflag.String(logLevelFlag, viper.GetString(logLevelFlag), "Agones Log level")
 	pflag.Duration(allocationBatchWaitTime, viper.GetDuration(allocationBatchWaitTime), "Flag to configure the waiting period between allocations batches")
 	pflag.String(podNamespace, viper.GetString(podNamespace), "namespace of current pod")
+	pflag.String(httpPort, viper.GetString(httpPort), "Port for the HTTP server. Defaults to 8080, can also use HTTP_PORT env variable")
```
Comment (zmerlynn) [inline cmd/controller/main.go:314]: If this is for webhooks, you only need this in `extensions`.

**Review by zmerlynn — APPROVED** [review]
Awesome! Thanks additionally for the cleanup of some remaining parameters that were clearly wrong after the `extensions` / `controller` split.

**Review by zmerlynn — APPROVED** [review]
Nice work! This just missed the 1.42 release (which we're prepping for right now), sorry for the slow review, but it'll be in the next.

**Comment by markmandel** [comment]
Gah - I didn't get to review this in time, my fault. This PR needs docs to round this work out. I just filed #3911 , and will discuss over there.

**Comment by zmerlynn** [comment]
> Gah - I didn't get to review this in time, my fault. This PR needs docs to round this work out. I just filed #3911 , and will discuss over there.

Shoot, sorry, you mentioned it in our internal chat and I forgot.

**Pull request merged**